server/src/main/java/org/cloudfoundry/identity/uaa/oauth/jwt/JwtClientAuthentication.java

```diff
@@ -65,4 +104,61 @@ private static HashMap<String, String> getJwtClientConfigurationElements(Object
     }
     return jwtClientConfiguration;
   }
+
+  public boolean validateClientJwt(Map<String, String[]> requestParameters, ClientJwtConfiguration clientJwtConfiguration, String clientId) {
+    if (GRANT_TYPE.equals(UaaStringUtils.getSafeParameterValue(requestParameters.get(CLIENT_ASSERTION_TYPE)))) {
+      try {
+        String clientAssertion = UaaStringUtils.getSafeParameterValue(requestParameters.get(CLIENT_ASSERTION));
+        if (!clientId.equals(getClientId(clientAssertion))) {
+          throw new BadCredentialsException("Wrong client_assertion");
+        }
+        JWT clientToken = JWTParser.parse(UaaStringUtils.getSafeParameterValue(requestParameters.get(CLIENT_ASSERTION)));
```

Comment:
```suggestion
        JWT clientToken = JWTParser.parse(clientAssertion);
```

Reply:
Can we reuse the `clientAssertion` variable here?

Reply:
done